sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -470,14 +470,36 @@ case class AlterTableAddPartitionCommand(
       CatalogTablePartition(normalizedSpec, table.storage.copy(
         locationUri = location.map(CatalogUtils.stringToURI)))
     }
-    catalog.createPartitions(table.identifier, parts, ignoreIfExists = ifNotExists)
+
+    // Hive metastore may not have enough memory to handle millions of partitions in single RPC.
+    // Also the request to metastore times out when adding lot of partitions in one shot.
+    // we should split them into smaller batches
+    val batchSize = 100
```

Comment:
can we make it configurable?

Reply:
Done.

Reply:
Reverted making config for batch size after some experiments and discussion with @srowen .

we noticed that different batch size doesn't change the overall performance. So we don't need to make this configurable. Probably thats the reason it was not made configurable in  [AlterTableRecoverPartitions earlier](https://github.com/apache/spark/blob/v2.4.4/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L740).

Reply:
Yeah I don't think it's worth yet another config if it's not going to have much meaning to set.

Reply:
instead of hard-coding this, we should still make it configurable. Let us add an internal conf.